Update README file and init changes syntax a little.

AutoViML opened the pull request.

AutoViML committed:
Update README file and init changes syntax a little.

AutoViML committed:
Update README file and init changes syntax a little.

ngupta23 committed:
merged local changes with giygub develop plus some cleanup based on linter recommendations.

ngupta23 committed:
cleaned up some tests

ngupta23 committed:
fixed some more tests

ngupta23 committed:
fixed more tests

ngupta23 committed:
updated all unit tests to the latest version of the code base.

ngupta23 commented on auto_ts/__init__.py:
```diff
@@ -136,10 +131,10 @@ def __init__(
         #####################################################################################################
         """
-
+        lag_period = 5
```
@AutoViML Why hard code this value? Should the user not be ale to decide this?

AutoViML replied:
At the moment: forecast_period is needed in model.predict() function. I felt that it did not belong in the auto_timeseries() object definition, hence I have just provided a placeholder value in object_definition. Will remove it once I find where all we have used forecast_period in the initialization object. I feel that it is not needed there. Once I find all the places it is used there and remove, then this lag_period will also be moved or removed. This is just a temporary situation. Hope this helps.

ngupta23 replied:
Hi @AutoViML 

It is used to decide the cross validation steps (in `fit`), so it is needed. The idea is that you provide it once when you initialize the object and then you don't have to specify it again during `fit` or `predict`. 

The only time you need to provide it during `predict` is when you need the forecast horizon to be different than the one used for training (which is rare). I just added that extra flexibility in there in case some one needed it, but by default the  `forecast_period` is not needed to be specified in `predict`.

Could you please revert it back. If you are concerned that it needs a default value, you can add it to the `__init__` call and set it to 1 (instead of 5).

Thanks!

ngupta23 committed:
converted VAR to a more modular initial structure

AutoViML commented:
Please take a look and let me know if we can push this.

AutoViML committed:
Merge pull request #22 from AutoViML/feature/nikhil

Merging to Synchronize + cleanup

AutoViML committed:
Updated init.py with forecast_period setting as 1.

AutoViML committed:
Updated init.py with forecast_period setting as 1.

AutoViML reviewed (commented):
Modified init.py with forecast_period set to 1.

AutoViML commented:
Nikhil:
Ok I have modified it based on my understanding - please review and let me
know. If not, you can change it too.
Thx
Ram

On Sun, Dec 20, 2020 at 11:02 AM Nikhil Gupta <notifications@github.com>
wrote:

> *@ngupta23* commented on this pull request.
> ------------------------------
>
> In auto_ts/__init__.py
> <https://github.com/AutoViML/Auto_TS/pull/21#discussion_r546395704>:
>
> > @@ -136,10 +131,10 @@ def __init__(
>          and Scikit-Learn ML. It will automatically select the BEST model which gives best score specified.
>          #####################################################################################################
>          """
> -
> +        lag_period = 5
>
> Hi @AutoViML <https://github.com/AutoViML>
>
> It is used to decide the cross validation steps (in fit), so it is
> needed. The idea is that you provide it once when you initialize the object
> and then you don't have to specify it again during fit or predict.
>
> The only time you need to provide it during predict is when you need the
> forecast horizon to be different than the one used for training (which is
> rare). I just added that extra flexibility in there in case some one needed
> it, but by default the forecast_period is not needed to be specified in
> predict.
>
> Could you please revert it back. If you are concerned that it needs a
> default value, you can add it to the __init__ call and set it to 1
> (instead of 5).
>
> Thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/AutoViML/Auto_TS/pull/21#discussion_r546395704>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMKBH6DVZRTM7KX6AMR5HTDSVYN2VANCNFSM4VB65ZBQ>
> .
>

ngupta23 committed:
Merge branch 'feature_ram' of https://github.com/AutoViML/Auto_TS into feature_ram

ngupta23 committed:
fixed issues with PR
